File: 1_easy/3001_4000/3201_3400/03364_minimum_positive_sum_subarray/solution.py
class Solution:
    def minimumSumSubarray(self, nums: List[int], l: int, r: int) -> int:
        min_positive_sum = -1
        n = len(nums)
        l -= 1
        r -= 1

        for i in range(n-l):
            curr_sum = 0
            for j in range(i, min(i + r, n - 1) + 1):
                curr_sum += nums[j]
                if j >= i + l and curr_sum > 0 and (curr_sum < min_positive_sum or min_positive_sum == -1):
                    min_positive_sum = curr_sum

        return min_positive_sum

File: 1_easy/3001_4000/3201_3400/03364_minimum_positive_sum_subarray/solution.cpp
class Solution {
public:
    int minimumSumSubarray(vector<int>& nums, int l, int r) {
        int min_positive_sum = -1, n = nums.size();
        l -= 1;
        r -= 1;

        for (int i{0}; i < n - l; ++i) {
            int sum = 0;
            for (int j{i}; j <= min(i + r, n -1); ++j) {
                sum += nums[j];
                if (j >= i + l && sum > 0 && (sum < min_positive_sum || min_positive_sum == -1)) {
                    min_positive_sum = sum;
                }
            }
        }

        return min_positive_sum;
    }
};
